src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -117,22 +152,26 @@ internal static string RemoveExtendedPrefix(string path)
             if (IsExtendedUnc(path))
                 return path.Remove(2, 6);
 
-            return path.Substring(4);
+            return path.Remove(0, 4);
         }
 
         /// <summary>
-        /// Removes the extended path prefix (\\?\) if present.
+        /// Remove the extended prefix from the given path. Return the type of path found.
         /// </summary>
-        internal static StringBuilder RemoveExtendedPrefix(StringBuilder path)
+        internal static ExtendedType RemoveExtendedPrefix(ref string path)
```

Comment:
> I assume that when you don't care about the out that local allocation doesn't get optimized away. Do you know?

What local allocation do you think is getting allocated with the signature one way and not the other way?